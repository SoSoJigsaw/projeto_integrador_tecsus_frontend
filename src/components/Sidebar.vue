<template>
  <!-- Menu lateral -->
  <div class="sidebar">
    <div class="logo">
      <p class="logo-text">
       PREDICTIVE <br>
       HEALTH</p>
    </div>

    <nav class="links">
      <RouterLink to="/dashboard">
      <img
        src="../assets/icons/dashboard.png"
        class="icon_img"
        height="50px"
      />  
      Dashboard</RouterLink>
      <RouterLink to="/audit-logs">
        <img
        src="../assets/icons/audit-logs.png"
        class="icon_img"
        height="50px"
        /> 
        Audit Logs</RouterLink>
      <RouterLink to="/health-prediction">
        <img
        src="../assets/icons/health-prediction.png"
        class="icon_img"
        height="45px"
        style="margin-left: 2px; margin-right: 2px;"
        /> 
        Predictions</RouterLink>
      <RouterLink to="/patients/:id">
        <img
        src="../assets/icons/patient-details.png"
        class="icon_img"
        height="50px"
        > 
        Your Details</RouterLink>
      <RouterLink to="/export-data/:id">
        <img
        src="../assets/icons/personal-data-export.png"
        class="icon_img"
        height="40px"
        style="margin-left: 5px; margin-right: 4px;"
        /> 
        Personal Data</RouterLink>
      <RouterLink to="/register-patient">
        <img
        src="../assets/icons/register-patient.png"
        class="icon_img"
        height="50px"
        /> 
        Register Patient</RouterLink>
      <RouterLink to="/patients">
        <img
        src="../assets/icons/show-patients.png"
        class="icon_img"
        height="45px"
        style="margin-right: 4px;"
        /> 
        Patients</RouterLink>
      <RouterLink to="/profile">
        <img
        src="../assets/icons/profile.png"
        class="icon_img"
        height="50px"
        /> 
        User Profile</RouterLink>
      <RouterLink to="/manage-users">
        <img
        src="../assets/icons/manage-users.png"
        class="icon_img"
        height="50px"
        /> 
        Manage Users</RouterLink>
      <RouterLink to="/register-user">
        <img
        src="../assets/icons/register-patient.png"
        class="icon_img"
        height="50px"
        /> 
        Register User</RouterLink>
      <RouterLink to="/change-password">
        <img
        src="../assets/icons/change-password.png"
        class="icon_img"
        height="50px"
        /> 
        Password</RouterLink>
      <RouterLink to="/consent-update/:id">
        <img
        src="../assets/icons/consent-update.png"
        class="icon_img"
        height="50px"
        /> 
        Consent Update</RouterLink>
      <RouterLink to="/">
        <img
        src="../assets/icons/login.png"
        class="icon_img"
        height="50px"
        /> 
        Login</RouterLink>
    </nav>
  </div>
</template>

<script lang="ts">
import { RouterLink } from 'vue-router'

export default {
  data() {
    return {
      animationDelay: false
    }
  },

  methods: {
    changeAnimationDelay() {
      this.animationDelay = true
    }
  }
}
</script>

<style>
@import '../assets/css/base.css';

.sidebar:not(.logo) {
  position: fixed;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: flex-start;
  gap: 2rem;
  top: 0;
  left: 0;
  width: calc(100vw - 80vw);
  height: 100vh;
  background-color: var(--azul-acizentado-0-5);
  padding-top: 1.5%;
  box-shadow: 0 4px 8px rgba(0, 0, 0, 0.3), 
              0 5px 4px rgba(39, 174, 96, 0.85);
  overflow-y: auto;
  scrollbar-width: none;
  -ms-overflow-style: none;
}

.sidebar::-webkit-scrollbar {
  display: none; 
}

.sidebar:hover {
  box-shadow: 0 6px 12px rgba(0, 0, 0, 0.4), 0 3px 6px rgba(39, 174, 96, 0.3);
  transition: box-shadow 0.3s ease;
}

.logo {
  display: absolute;
  background-color: transparent;
  height: 100%;
  width: auto;
  font-size: 2rem;
  font-weight: 600;
  color: var(--branco-auxiliar);
  margin-right: 30%;
}

.links {
  width: calc(25vw - 1.5vw);
  display: flex;
  flex-direction: column;
  gap: 0.5rem;
  font-size: 1.3rem;
  text-align: center;
  font-weight: 800;
  margin-left: calc(20vw - 18.5vw);
}

.links a.router-link-exact-active,
.links a.router-link-exact-active i {
  transition: all 0.5s;
  color: var(--verde-contraste);
  background-color: transparent;
}

.links a.router-link-exact-active i {
  transition: all 0.5s;
}

.links a.router-link-exact-active:hover {
  transition: all 0.5s;
  padding-left: 0.8%;
}

.links a {
  transform: all 0.5s;
  color: var(--verde-contraste-0-8);
  text-decoration: none;
  display: flex;
  flex-direction: row;
  justify-content: flex-start;
  align-items: center;
  margin-left: 8%;
}

.links a:hover:not(nav a.router-link-exact-active:hover) {
  transition: all 0.5s;
  padding-left: 2%;
}

.icon_img {
  padding-right: 1%;
}
</style>
