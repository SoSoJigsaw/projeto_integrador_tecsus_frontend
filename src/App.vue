<script lang="ts">
import { RouterView } from 'vue-router';
import Sidebar from '@/components/Sidebar.vue';


export default {

  data() {
    return {

    }
  },

  components: {
    Sidebar
  },

}
</script>

<template>
  <div class="app-container">
    <!-- Menu lateral -->
    <Sidebar/> 
  </div>

  <transition name="fade" mode="out-in">
    <router-view></router-view>
  </transition>
</template>

<style scoped>
@import './assets/base.css';

/* estilização dos elementos do single page */
.app-container {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: flex-start;
}

</style>
