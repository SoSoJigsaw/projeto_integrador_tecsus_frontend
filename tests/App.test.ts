import { mount } from '@vue/test-utils';
import App from '../src/App.vue';
import Sidebar from '../src/components/Sidebar.vue';
import { createRouter, createWebHistory } from 'vue-router';
import { describe, it, expect, vi } from 'vitest';

const router = createRouter({
  history: createWebHistory(),
  routes: [{ path: '/', component: App }],
});

describe('App.vue', () => {
  it('should mount the component correctly', () => {
    const wrapper = mount(App, {
      global: {
        plugins: [router],
        stubs: {
          'router-view': true,
          'router-link': true,
        },
      },
    });
    expect(wrapper.exists()).toBe(true);
  });

  it('should render the Sidebar component', () => {
    const wrapper = mount(App, {
      global: {
        plugins: [router],
        stubs: {
          'router-view': true,
          'router-link': true,
        },
      },
    });
    expect(wrapper.findComponent(Sidebar).exists()).toBe(true);
  });

  it('should include a RouterView component', () => {
    const wrapper = mount(App, {
      global: {
        plugins: [router],
        stubs: {
          'router-view': true,
          'router-link': true,
        },
      },
    });
    expect(wrapper.findComponent({ name: 'RouterView' }).exists()).toBe(true);
  });

  it('should have a main container with specific style', () => {
    const wrapper = mount(App, {
      global: {
        plugins: [router],
        stubs: {
          'router-view': true,
          'router-link': true,
        },
      },
    });
    expect(wrapper.find('div.app-container').exists()).toBe(true);
  });

  it('should apply scoped styles', () => {
    const wrapper = mount(App, {
      global: {
        plugins: [router],
        stubs: {
          'router-view': true,
          'router-link': true,
        },
      },
    });
    const hasScopedStyles = wrapper.html().includes('data-v-');
    expect(hasScopedStyles).toBe(true);
  });

  it('should render without errors with default props', () => {
    const wrapper = mount(App, {
      global: {
        plugins: [router],
        stubs: {
          'router-view': true,
          'router-link': true,
        },
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should clean up when the component is destroyed', () => {
    const wrapper = mount(App, {
      global: {
        plugins: [router],
        stubs: {
          'router-view': true,
          'router-link': true,
        },
      },
    });
    wrapper.unmount();
    expect(wrapper.exists()).toBe(false);
  });

  it('should handle navigation correctly', async () => {
    const wrapper = mount(App, {
      global: {
        plugins: [router],
        stubs: {
          'router-view': true,
          'router-link': true,
        },
      },
    });
    await router.push('/');
    await wrapper.vm.$nextTick();
    expect(wrapper.findComponent(App).exists()).toBe(true);
  });
});
