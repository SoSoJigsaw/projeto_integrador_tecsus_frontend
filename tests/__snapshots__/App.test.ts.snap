// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App.vue > should render without errors with default props 1`] = `
"<div data-v-7a7a37b1="" class="app-container">
  <!-- Menu lateral -->
  <!-- Menu lateral -->
  <div data-v-7a7a37b1="" class="sidebar">
    <div class="logo"><img src="/src/assets/logo/tecsus.png" class="logo_img" style=""></div>
    <nav class="links">
      <router-link-stub to="/" replace="false" custom="false" ariacurrentvalue="page"></router-link-stub>
      <router-link-stub to="/reports" replace="false" custom="false" ariacurrentvalue="page"></router-link-stub>
      <router-link-stub to="/alerts" replace="false" custom="false" ariacurrentvalue="page"></router-link-stub>
      <router-link-stub to="/importar-dados" replace="false" custom="false" ariacurrentvalue="page"></router-link-stub>
    </nav>
  </div>
</div>
<router-view-stub data-v-7a7a37b1=""></router-view-stub>"
`;
